libraries/AP_Camera/AP_Camera_Tracking.cpp

```diff
@@ -0,0 +1,87 @@
+#include "AP_Camera_Tracking.h"
+
+#if AP_CAMERA_TRACKING_ENABLED
+
+extern const AP_HAL::HAL& hal;
+
+// set tracking to none, point or rectangle (see TrackingType enum)
+// if POINT only then top_left is the point
+// top_left,bottom_right are in range 0 to 1.  0 is left or top, 1 is right or bottom
+bool AP_Camera_Tracking::set_tracking(TrackingType tracking_type, const Vector2f& top_left, const Vector2f& bottom_right, uint8_t tracking_device_sysid, uint8_t tracking_device_compid, mavlink_camera_information_t _cam_info)
+{
+    // if we don't support the required tracking then return
+    switch (tracking_type) {
+        case TrackingType::TRK_NONE:
+            break;
+        case TrackingType::TRK_POINT:
+            if (!(_cam_info.flags & CAMERA_CAP_FLAGS_HAS_TRACKING_POINT)) {
+               return false;
+            }
+            break;
+        case TrackingType::TRK_RECTANGLE:
+            if (!(_cam_info.flags & CAMERA_CAP_FLAGS_HAS_TRACKING_RECTANGLE)) {
+               return false;
+            }
+            break;
+    }
+
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO,"Tracking: New Tracking request");
+    uint8_t proxy_device_compid = tracking_device_compid; 
+    uint8_t proxy_device_sysid;
```

Comment:
Needs initialising to avoid compile error on macOS.

```c++
= tracking_device_compid; 
    uint8_t proxy_device_sysid {};
```

Reply:
Thanks